MIGRATING.md

```diff
@@ -4,6 +4,119 @@ This guide describes the steps to upgrade dd-trace from a major version to the
+on `http.request` for incoming requests should be updated to `web.request`.
+
+With this change, both operation names also appear under the main service name
+and are no longer split between the server service name and a separate client
+service name suffixed with `-http-client`.
+
+### gRPC operation name change
+
+The gRPC integration now uses `grpc.server` for incoming requests and
+`grpc.client` for outgoing requests. Any
+[monitor](https://docs.datadoghq.com/monitors/create/types/apm/?tab=apmmetrics)
+on `grpc.request` should be updated to one of these.
+
+With this change, both operation names also appear under the main service name
+and are no longer split between the server service name and a separate client
+service name suffixed with `-http-client`.
+
+### Removal of `fs` integration
+
+The `fs` integration was removed as it was originally added without an actual
+use case, and it's been problematic ever since it's noisy, the output is
+confusing when using streams, errors that are handled higher in the stack end up
+being captured, etc.
+
+If you had any use for file system instrumentation, please let us know so we can
+provide an alternative.
+
+### Event emitter binding
+
+It's no longer possible to bind event emitters using
+`tracer.scope().bind(emitter)`. This was historically added mostly for
+internal use, and changes to context propagation over the years made this
+unnecessary, both internally and externaly. If this is used anywhere, the call
+will simply be ignored and no binding will occur.
+
+To bind all listeners for an event, instead wrap the call to `emit`:
+
+```js
+tracer.scope().activate(span, () => {
+  emitter.emit('event')
+})
+```
+
+To bind individual listeners, bind the listener function directly:
+
+```js
+emitter.on('event', tracer.scope().bind(listener, span))
+```
+
+Binding promises is still available but is deprecated with this release and will
+be removed as well in 4.0 since native promises are already bound properly in
+Node.
+
+### Removed APIs
+
+The following APIs have been deprecated for a long time and have now been
+completely removed:
+
+- `tracer.currentSpan()`
+- `tracer.bindEmitter()`
+
+Since these have not been recommended nor publicly documented for years at this
+point, there should be no impact as no application is expected to be using them.
+
+### CI Visibility new entrypoints
+
+#### Cypress
+`dd-trace/cypress/plugin` and `dd-trace/cypress/support` are removed, so you won't 
+be able to use them for your `cypress` instrumentation. Use `dd-trace/ci/cypress/plugin`
+and `dd-trace/ci/cypress/support` instead for your plugin and support configuration 
+respectively. 
+
+#### Jest
+The use of `'dd-trace/ci/jest/env'` in [`testEnvironment`](https://jestjs.io/docs/configuration#testenvironment-string)
+is no longer supported. 
+The way to instrument your `jest` tests now is by passing `NODE_OPTIONS='-r dd-trace/ci/init'`
```

Comment:
`passing X to the process` seems like odd wording for defining an environment variable. I could imagine newbies that don't know much about environment variables yet thinking that should be passed as an argument somehow. Could probably be more clear about exactly how to use that.

Reply:
Clarified this a bit.

Reply:
If it's still not 100% clear we can always improve it later.